physioqc/interfaces/visualizations.py

```diff
@@ -0,0 +1,197 @@
+from typing import List, Literal, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import peakdet as pk
+
+from ..metrics import multimodal
+
+
+def check_create_figure(
+    ax: plt.axes = None, figsize=(10, 5)
+) -> Tuple[plt.figure, plt.axes]:
+    """Helper function to check if a new figure should be created or to further
+    plot on axes.
+
+    Parameters
+    ----------
+    ax : plt.axes, optional
+        Check is axes object is present, if not creates new figure, by default None
+    figsize : tuple, optional
+        Size of the new figure (if created), by default (10, 5)
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1, figsize=figsize)
+    else:
+        fig = None
+
+    return fig, ax
+
+
+def plot_raw_data(phys: pk.Physio, ax: plt.axes = None) -> Tuple[plt.figure, plt.axes]:
+    """Plots the raw data using peakdet.
+    Parameters
+    ----------
+    phys : pk.Physio
+        The data object, used for plotting.
+    ax : plt.axes, optional
+        axes object to use for plotting, by default None
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+    fig, ax = check_create_figure(ax, figsize=(7, 5))
+
+    ax = pk.plot_physio(phys, ax=ax)
+
+    return fig, ax
+
+
+def plot_average_peak(
+    phys: pk.Physio,
+    window: List = [-3, 3],
+    target_fs: float = 1000.0,
+    peak_dist: float = 1.0,
+    peak_thr: float = 0.1,
+    plot_mode: Literal["traces", "ci", "auto"] = "auto",
+    ax: plt.axes = None,
+) -> Tuple[plt.figure, plt.axes]:
+    """_summary_
+
+    Parameters
+    ----------
+    phys : pk.Physio
+        data object, used for plotting.
+    window : List, optional
+        window size around the peak in s, by default [-3, 3]
+    target_fs : float, optional
+        sampling rate for plotting and peak detection, by default 1000.0
+    peak_dist : float, optional
+        distance for peak detection, by default 1.0
+    peak_thr : float, optional
+        threshold for peak detection, by default 0.1
+    plot_mode : Literal["traces", "ci"], optional
+        to plot traces or standard deviation around signal, by default 'auto'
+    ax : plt.axes, optional
+        axes object to use for plotting, by default None
+
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+
+    Raises
+    ------
+    ValueError
+        If an inapropriate plot type is chosen.
+    """
+    if plot_mode not in ["traces", "ci", "auto"]:
+        raise ValueError('Plot mode has to be in ["traces", "ci", "auto"]')
+
+    if phys.fs > 1_000:
+        #  This should probably go to the logs, or warning
+        print(f"Fs = {phys.fs} Hz too high, resampling to {target_fs} Hz")
+        phys = pk.operations.interpolate_physio(phys, target_fs, kind="linear")
```

Comment:
Could consider raising an error if target_fs > 1000 Hz.

Reply:
As with the sentiment above, I think it should be possible to re-sample to larger than 1000 Hz if one really wants to and know what they are doing.

The warning / message is probably misleading. I think I would change it to phys.fs > target_fs  and display the message: "Fs = xx Hz higher than target_fs, resampling to target_fs or something.

For plotting purposes, I still think it makes sense to enforce an upper limit for the sampling rate - and 1000 Hz seemed reasonable enough for me, at the Brainhack ...